source/Ship.cpp

```diff
@@ -2622,6 +2659,31 @@ double Ship::MaximumHeat() const
 
 
 
+bool Ship::IsCloaked() const
+{
+	return Cloaking() == 1.;
+}
+
+
+
+double Ship::CloakingSpeed() const
+{
+	return attributes.Get("cloak") + attributes.Get("cloak by mass") * 1000. / Mass();
+}
+
+
+
+bool Ship::Phases(Projectile projectile) const
```

Comment:
This is making a copy of the projectile, almost certainly not what you intended - but again, why is this needed?

Reply:
because functions like this make the code clearer and cleaner? idk xD